gulpfile.js

```diff
@@ -106,21 +106,36 @@ gulp.task('compile_test', function() {
 });
 
 gulp.task('copyTypings', function() {
-  return gulp.src(['src/index.d.ts', 'src/firebase-namespace.d.ts'])
+  return gulp.src([
+    'src/index.d.ts',
+    'src/firebase-namespace.d.ts',
+    'src/**/protos/*.d.ts',
+  ])
     // Add header
     .pipe(header(banner))
     .pipe(gulp.dest(paths.build))
 });
 
-gulp.task('compile_all', gulp.series('compile', 'copyTypings', 'compile_test'));
+gulp.task('copyJSON', function() {
+  return gulp.src([
+    'src/**/protos/*.json',
+    // This isn't ideal, but doing something like
+    // 'src/generated/**/*.json' results in incorrect paths in the /lib dir
+    'src/**/*.json'
```

Comment:
Can we say `src/generated/**/*.json` instead?

Reply:
Ah I see. It ends up copying them into `lib/messaging` instead of `lib/generated/messaging`. I guess it only matches path segments after `src/generated` when the pattern is specified this way. Then I think the way you've implemented this is correct.

Reply:
Yeah, doing the more general wildcard is less "comfortable" than doing the specific `src/generated/**/*.json`, but it's (as far as I could tell) the only way for the generated path to be structured properly.